src/nvim/api/tabpage.c

```diff
@@ -14,6 +16,27 @@
 # include "api/tabpage.c.generated.h"
 #endif
 
+/// Opens a new tabpage with a single window
+///
+/// @param buffer   Buffer handle, or 0 for current buffer
+/// @param enter    Boolean, whether to enter the new tabpage
+/// @param[out] err Error details, if any
+/// @return Handle to newly created tabpage
+Tabpage nvim_open_tabpage(Buffer buffer, Boolean enter, Error *err)
```

Comment:
Ignoring whether this API is wanted... we almost always want to accept `opt` and return `Dict`, unless we are 100% certain we won't need to expand the features in the future.
```suggestion
Dict nvim_open_tabpage(Buffer buffer, Boolean enter, Dict opts, Error *err)
```

Reply:
Good to know, I'll remember that in the future :)

re: whether this API is wanted

This was largely an experiment and I don't have a real use case for it, so no worries if it's not wanted.